ledger/simulation/tracer.go

```diff
@@ -245,6 +251,32 @@ func (tracer *evalTracer) makeOpcodeTraceUnit(cx *logic.EvalContext) OpcodeTrace
 	return OpcodeTraceUnit{PC: uint64(cx.PC())}
 }
 
+func (o *OpcodeTraceUnit) appendDeletedStackValue(cx *logic.EvalContext, tracer *evalTracer) {
+	tracer.stackChangeExplanation = cx.NextStackChange()
+
+	stackCopy := cx.Stack()
+	stackHeight := len(stackCopy)
+	stackHeightAfterDeletion := stackHeight - tracer.stackChangeExplanation.Deletions
+	tracer.stackHeightAfterDeletion = stackHeightAfterDeletion
+
+	if tracer.stackChangeExplanation.Deletions == 0 {
+		return
+	}
+	for i := stackHeightAfterDeletion; i < stackHeight; i++ {
+		if stackCopy[i].TEALType() == basics.TealBytesType {
+			o.Deleted = append(o.Deleted, TealValue{
+				Type:  basics.TealBytesType,
+				Bytes: string(stackCopy[i].Bytes),
+			})
+		} else {
+			o.Deleted = append(o.Deleted, TealValue{
+				Type: basics.TealUintType,
+				Uint: stackCopy[i].Uint,
+			})
+		}
```

Comment:
I think that by using `basics.TealType` this becomes:
```suggestion
		o.Deleted = append(o.Deleted, stackCopy[i].toTealValue())
```